CHANGELOG.md

```diff
@@ -3,6 +3,11 @@ All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/) and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
+## [6.1.3] - 2020-05-11
+### Changed
+- Set min/max size of HMS thread pool based on memory.  Max will be set to 1 connection for every 2MB RAM.  Min will be 0.25% of max.  This will prevent large HMS instances from not having enough threads/connections available.
```

Comment:
Just to make sure I understand - so this is based on how much ram has been assigned to the HMS, not what's available on the machine right? So if we assigned 4GB of RAM to the HMS we'd get ~2000 threads as the maximum? How did you decide on this link between the two values?

Reply:
This is actually based on how much ram as been assigned to the instance (be it EKS or ECS).  However, we give 90% of that ram to HMS (https://github.com/ExpediaGroup/apiary-metastore-docker/blob/master/files/startup.sh#L165-L168).

After investigating on-prem HMS where we have 100K max threads configured, and after looking at this Apache Hive Jira (https://issues.apache.org/jira/browse/HIVE-8666), we decide it would be safe to dedicate up to 50% of the instance's ram to threads and connections when it is busy.  When it is not busy, the thread pool will shrink to the minimum size, which we are configuring at 1/4 of the max, or 12.5% of available ram.